File: presentation/q2/simple_linear_regression.R
install.packages("caTools")
library(caTools)

# Read data into the 'salary_data' variable
salary_data = read.csv("/Users/tunwa/Documents/coding-files/cpe322/presentation/salary.csv", sep=",", header = TRUE)
View(salary_data)

# Generates a randomized sample of size (number of data samples) containing 0s and 1s using the ratio 8:2
set.seed(3)
id <- sample(0:1, nrow(salary_data), prob = c(0.8, 0.2), replace = TRUE)
# Data samples which map to the 0 value from the sample are assigned to the training dataset
training_dataset = salary_data[id == 0,]
View(training_dataset)
# Data samples which map to the 1 value from the sample are assigned to the testing dataset
testing_dataset = salary_data[id == 1,]
View(testing_dataset)

lm.r = lm(formula = Salary ~ YearsExperience, data = training_dataset)

test_pred <- predict(lm.r, testing_dataset)

test_pred

summary(lm.r)

File: mine/presentation/q1/simple_linear_regression.py
from pandas import read_csv

from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, mean_absolute_error

# Read data from a sample diabetes dataset
salary_data = read_csv("/Users/tunwa/Documents/coding-files/cpe322/presentation/salary.csv")

# Split data into features (X) and labels (y)
X = salary_data[salary_data.columns[1:2]].values
y = salary_data[salary_data.columns[-1]].values

# Split training and test data using the ratio 8:2
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8, random_state=25)

# Instantiate the model
lr_model = LinearRegression()

# Train the model
lr_model.fit(X_train, y_train)

# Make predictions using the model
y_pred = lr_model.predict(X_test)

# - Evaluate the performance of the model
print("Simple Linear Regression Results\n")
print(f"Intercept: \n {lr_model.intercept_}")
print(f"Coefficient: \n {lr_model.coef_}")
print(f"Mean absolute error: \n {mean_squared_error(y_test, y_pred)}")
print(f"Mean squared error: \n {mean_absolute_error(y_test, y_pred)}")